src/test/ui/borrowck/two-phase-across-loop.stderr

```diff
@@ -2,7 +2,21 @@ error[E0499]: cannot borrow `foo` as mutable more than once at a time
   --> $DIR/two-phase-across-loop.rs:17:22
    |
 LL |         strings.push(foo.get_string());
-   |                      ^^^^^^^^^^^^^^^^ `foo` was mutably borrowed here in the previous iteration of the loop
+   |         -------------^^^^^^^^^^^^^^^^-
+   |         |            |
+   |         |            `foo` was mutably borrowed here in the previous iteration of the loop
+   |         first borrow later used here
```

Comment:
I think these labels could probably be special-cased for this case to provide a better diagnostic, something like:

```suggestion
   |         -------------^^^^^^^^^^^^^^^^-
   |         |            |
   |         |            `foo` is mutably borrowed here in a loop iteration..
   |         ..and that borrow lives into the next loop iteration, preventing another mutable borrow
```

But I'm not sure how well a label like that would work for every case that this changes.

Reply:
Agreed. For preserving the wording for other cases and not changing too many code, I changed explanation part for this case using your suggested wording with a small modification 
```
 LL |         strings.push(foo.get_string());
-   |                      ^^^^^^^^^^^^^^^^ `foo` was mutably borrowed here in the previous iteration of the loop
+   |         -------------^^^^^^^^^^^^^^^^-
+   |         |            |
+   |         |            `foo` was mutably borrowed here in the previous iteration of the loop
+   |         that borrow lives into the next loop iteration, preventing another mutable borrow because the borrow later used here
```
Any comments?

Reply:
I think you should feel free to thread some `IsInLoop::Yes`-type context into `explain_why_borrow_contains_point` so that you can adjust the message rather than adding to the string, since that's harder to do with diagnostic translation (#100717) anyway.

Reply:
I found in the original code, `BorrowExplanation::UsedLaterInLoop` in `explain_why_borrow_contains_point` is never being used, since the function `is_use_in_later_iteration_of_loop` will never return `true`. I suspect that since we are using NLL now, this portion of the code is not being relevant anymore.
I modified the code by reusing `BorrowExplanation::UsedLaterInLoop` for the case, and removed 125 line of code which related to the unused `is_use_in_later_iteration_of_loop` function. This gives use more freedom to adjust the message for "used later in loop" case only and not affect other cases.